Source/ASCollectionView.mm

```diff
@@ -916,7 +916,9 @@ - (CGSize)collectionView:(UICollectionView *)collectionView layout:(UICollection
   ASCellNode *cell = [self nodeForItemAtIndexPath:indexPath];
   if (cell.shouldUseUIKitCell) {
     if ([_asyncDelegate respondsToSelector:@selector(collectionView:layout:sizeForItemAtIndexPath:)]) {
-      return [(id)_asyncDelegate collectionView:collectionView layout:collectionViewLayout sizeForItemAtIndexPath:indexPath];
+      CGSize size = [(id)_asyncDelegate collectionView:collectionView layout:collectionViewLayout sizeForItemAtIndexPath:indexPath];
+      cell.style.preferredSize = size;
```

Comment:
Should we do the same [here](https://github.com/facebook/AsyncDisplayKit/pull/3178/files?diff=split#diff-8b82b1556d1293f6f9789beb236c9271R936) and [here](https://github.com/facebook/AsyncDisplayKit/pull/3178/files?diff=split#diff-8b82b1556d1293f6f9789beb236c9271R951)?